fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/EraseResourceDAO.java

```diff
@@ -0,0 +1,351 @@
+            try (PreparedStatement stmt = getConnection().prepareStatement(UPDATE_RESOURCE_PAYLOAD)) {
+                stmt.setLong(1, logicalResourceId);
+                stmt.setInt(2, eraseDto.getVersion());
+                int count = stmt.executeUpdate();
+                total += count;
+            } catch (SQLException x) {
+                LOG.log(Level.SEVERE, UPDATE_RESOURCE_PAYLOAD, x);
+                throw translator.translate(x);
+            }
+            eraseRecord.setTotal(total);
+            eraseRecord.setStatus(ResourceEraseRecord.Status.VERSION);
+            return;
+        }
+
+        // Step 2: Delete All Versions from Resources Table
+        // Create the prepared statement to delete Resource Versions in chunks
+        // Implementation note: fetch must be the last part of the sub-select
+        final String DELETE_ALL_VERSIONS =
+                "DELETE FROM " + resourceType + "_RESOURCES" +
+                        "    WHERE LOGICAL_RESOURCE_ID = ?";
+        try (PreparedStatement stmt = getConnection().prepareStatement(DELETE_ALL_VERSIONS)) {
+            stmt.setLong(1, logicalResourceId);
+            total = stmt.executeUpdate();
+        } catch (SQLException x) {
+            LOG.log(Level.SEVERE, DELETE_ALL_VERSIONS, x);
+            throw translator.translate(x);
+        }
+
+        // Step 3: Delete from parameters tables
+        deleteFromAllParametersTables(resourceType, logicalResourceId);
+
+        // Step 4: Delete from Logical Resources table
+        final String DELETE_LOGICAL_RESOURCE =
+                "DELETE FROM " + resourceType + "_LOGICAL_RESOURCES WHERE LOGICAL_RESOURCE_ID = ?";
+        try (PreparedStatement stmt = getConnection().prepareStatement(DELETE_LOGICAL_RESOURCE)) {
+            stmt.setLong(1, logicalResourceId);
+            int count = stmt.executeUpdate();
+            LOG.fine(() -> "Count of Resource_LR deleted is " + count);
+        } catch (SQLException x) {
+            LOG.log(Level.SEVERE, DELETE_LOGICAL_RESOURCE, x);
+            throw translator.translate(x);
+        }
+
+        // Step 5: Delete from Global Logical Resources
+        final String GLOBAL_DELETE_LOGICAL_RESOURCE =
+                "DELETE FROM LOGICAL_RESOURCES WHERE LOGICAL_RESOURCE_ID = ? AND RESOURCE_TYPE_ID = ?";
+        try (PreparedStatement stmt = getConnection().prepareStatement(GLOBAL_DELETE_LOGICAL_RESOURCE)) {
+            stmt.setLong(1, logicalResourceId);
+            stmt.setLong(2, resourceTypeId);
+            int count = stmt.executeUpdate();
+            LOG.fine(() -> "Count of LR deleted is " + count);
+        } catch (SQLException x) {
+            LOG.log(Level.SEVERE, GLOBAL_DELETE_LOGICAL_RESOURCE, x);
+            throw translator.translate(x);
+        }
+
+        // Step 6: Delete from resource_change_log
+        final String CL_DELETE =
+                "DELETE FROM resource_change_log WHERE LOGICAL_RESOURCE_ID = ? AND RESOURCE_TYPE_ID = ?"
```

Comment:
I thought the plan was to not delete from the resource-change-log, but instead inject a new record to allow a downstream system the ability to sync-up.

Reply:
It's not possible to update without leaving the dangling logical_resources as the logicalId means nothing at this point in the code.As the Resource_change_log would require us to leave the logical_resources entry. 

```
        query.append(" SELECT c.resource_id, rt.resource_type, lr.logical_id, c.change_tstamp, c.version_id, c.change_type")
            .append("  FROM ")
            .append(schemaName).append(".resource_change_log  c, ")
            .append(schemaName).append(".logical_resources   lr, ")
            .append(schemaName).append(".resource_types      rt  ")
            .append(" WHERE lr.logical_resource_id = c.logical_resource_id ")
            .append("   AND rt.resource_type_id = c.resource_type_id ")
```

I don't think this is ideal, and would require a db change to integrate the logical_id into the table, which I don't think is ideal. 

I think the right thing is to alert folks to the business logic in the delete, and remove the resource_change_log entry.

Reply:
The above delete statement unfortunately isn't practical - it isn't supported by an index. If you need to delete records from the resource_change_log you need to use the resource_id. This is version-specific, which I think is also something you need.

Reply:
I've updated to use the resource_id

Reply:
Got what you mean... I've updated the code.